### ISSUE TITLE
NET-1665 Fix S1075 FN: Support tuple

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#8169]
S1075 does not raise issues on tuple:
```cs
(string uri1, string uri2) = ("C:/test.txt", "C:/test.txt"); // FN
```
is conceptually equivalent to:
```cs
string uri1 = "C:/test.txt"; // Noncompliant
string uri2 = "C:/test.txt"; // Noncompliant
```
which raises the issue for both.

Edge cases should also be considered:
```cs
(string uri1, string) a = ("C:/test.txt", "C:/test.txt"); // FN on the first
(string uri1, string) uri = ("C:/test.txt", "C:/test.txt"); // FN on both, because uri is the name of the tuple
(string b, (string uri, string c)) a = ("C:/test.txt", ("C:/test.txt", "C:/test.txt")); // FN only for uri
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1665](https://sonarsource.atlassian.net/browse/NET-1665)

[NET-1665]: https://sonarsource.atlassian.net/browse/NET-1665?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
